### PR TITLE
Add support for ESP32-H2

### PR DESCRIPTION
**Pull request by dwhitlow**
Feedback from #1347 requested that H2 and C2 support be split up into separate PRs. I've created this separate PR based on @laur123 's work and his fork of this repo.

Please let me know if any further work is necessary to get this merged.

### CODE REVIEW

File: boards/esp32-h2-devkitm-1.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "build": {
+    "core": "esp32",
+    "f_cpu": "96000000L",
```
Comment (Jason2866) [inline boards/esp32-h2-devkitm-1.json:4]: Is not fully correct. Missing an entry. Should be:
```
    "core": "esp32",
    "f_cpu": "96000000L",
    "f_flash": "64000000L",
    "f_image": "48000000L",
    "flash_mode": "qio",
    "mcu": "esp32h2",
    "variant": "esp32h2"
```

Comment (dwhitlow) [reply boards/esp32-h2-devkitm-1.json:4]: Thanks @Jason2866 ! I'll push these changes shortly. Anything else I should update or add to get this approved?

Comment (Jason2866) [reply boards/esp32-h2-devkitm-1.json:4]: I am not related to Platformio. A user like you. Providing now and than a PR ;-)

**Comment by dwhitlow** [comment]
@valeros apologies for the ping, but is there a process I need to follow to get this reviewed by a maintainer? I was unable to find any sort of contribution guidelines or documentation.

**Comment by raphael-bmec-co** [comment]
@robertlipe 

> PlatformIO is seemingly not merging even user-contributed PRs for ESP32 as a negotiation tactic

Are they really doing this? 😠

**Comment by raphael-bmec-co** [comment]
Right. I've been involved in the ongoing discussions. These seemed to be progressing with a degree of good faith. Blocking pull requests however is _disappointing_ behaviour. Good to see things are moving ahead, l hope the community has enough steam to keep these forks going 🤞 Will contribute what and where we can.

**Comment by dwhitlow** [comment]
I'm going to close this PR and focus on the community-maintained fork. Thanks @robertlipe and esphome community members for pointing out this alternative!

**Pull request closed without merging**